src/dotty/tools/dotc/transform/LazyVals.scala

```diff
@@ -124,7 +121,12 @@ class LazyVals extends MiniPhaseTransform with SymTransformer {
 
 
   override def transformStats(trees: List[tpd.Tree])(implicit ctx: Context, info: TransformerInfo): List[tpd.Tree] = {
-    val (holders, stats) = trees.partition { _.symbol.flags == containerFlags}
+    val (holders, stats) =
+      atGroupEnd { implicit ctx: Context =>
+        trees.partition {
+          _.symbol.flags == containerFlags
```

Comment:
What is this supposed to achieve? It would be good to have a comment why the statements are re-ordered.

In fact, I doubt there are re-ordered at all, since the comparison is using == on flags. Would the symbol not at least have flag Touched set? Generally, comparing flags with == is extremely fragile.